pkg/applicationserver/io/packages/loragls/v3/api/objects.go

```diff
@@ -85,30 +87,65 @@ type SingleFrameRequest struct {
 	Frame    Frame     `json:"frame"`
 }
 
-// BuildSingelFrameRequest builds a SingleFrameRequest from the provided metadata.
-func BuildSingleFrameRequest(metadata []*ttnpb.RxMetadata) *SingleFrameRequest {
-	r := &SingleFrameRequest{}
-	for _, m := range metadata {
-		if m.Location == nil {
-			continue
-		}
-		var tdoa *uint64
-		if m.FineTimestamp != 0 {
-			tdoa = &m.FineTimestamp
-		}
-		r.Gateways = append(r.Gateways, Gateway{
-			GatewayID: m.GatewayID,
+func parseRxMetadata(ctx context.Context, m *ttnpb.RxMetadata) (Gateway, Uplink) {
+	gtwUID := unique.ID(ctx, m.GatewayIdentifiers)
+	var tdoa *uint64
+	if m.FineTimestamp != 0 {
+		tdoa = &m.FineTimestamp
+	}
+	return Gateway{
+			GatewayID: gtwUID,
 			Latitude:  m.Location.Latitude,
 			Longitude: m.Location.Longitude,
 			Altitude:  float64(m.Location.Altitude),
-		})
-		r.Frame = append(r.Frame, Uplink{
-			GatewayID: m.GatewayID,
+		}, Uplink{
+			GatewayID: gtwUID,
```

Comment:
Do they need to know this, i.e. are we required to leak the UID? If not, let's refrain from doing so. I think they only need these values to be unique per request to correlate them in the algorithm, but they don't have to be stable over time and certainly not include potentially sensitive information (customer name, site names etc).